monai/data/dataset.py

```diff
@@ -103,6 +103,10 @@ class PersistentDataset(Dataset):
     If passing slicing indices, will return a PyTorch Subset, for example: `data: Subset = dataset[1:4]`,
     for more details, please check: https://pytorch.org/docs/stable/data.html#torch.utils.data.Subset
 
+    The transforms which are supposed to be cached must inherrit `monai.transforms.Transform` or its subclass
+    but should not be `Randomizable`. This dataset will cache until the first transform that is subclass of
+    `Randomizable` or is not subclass of `Transform`.
```

Comment:
```suggestion
    The transforms which are supposed to be cached must implement the `monai.transforms.Transform` 
    interface and should not be `Randomizable`. This dataset will cache the outcomes before the first
   `Randomizable` `Transform` within a `Compose` instance.
```

Reply:
Hi @wyli ,

Do you know how to **sign** the commit if I click your commit suggestion online directly?
Otherwise, it has DCO failure.

Thanks in advance.

Reply:
there's a discussion here... https://github.com/todogroup/gh-issues/issues/50#issuecomment-330144236

Reply:
Thanks!